### ISSUE TITLE
Can't find Error Event

### ISSUE THREAD

**Issue by Crisfole** [ractivejs/ractive#1919]
Is there an error event I can hook [rollbar](https://rollbar.com/) or any other global error handler up to?  Right now I'm not getting any feedback when an error reaches the console in production.

**Comment by martypdx** [comment]
@Crisfole a specific example would be helpful so I can get my head around which class of errors you'd want to propagate.

**Comment by Crisfole** [comment]
I want to propagate anything that gets triggered on a top level Ractive (basically: if it has no parent).

When I run:

    window.ractive = new Ractive({ template: "#template", el: "body" });

or whatever, I'd like to be able to say:

    window.ractive.on('error', function(errorInfo) {
        rollbar.reporterrror(errorInfo); // made up syntax, but it makes the point.
    })

**Comment by martypdx** [comment]
Oh, add it as a event hook. That would actually be doable. :)

**Comment by martypdx** [comment]
With current event bubbling, what you wrote would work on top-level, but you could also do:

```js
ractive.on('error *.error', function() {
    // all instance errors including components
})

**Comment by Crisfole** [comment]
So it already exists?

**Comment by martypdx** [comment]
@Crisfole no, I marked as enhancement. The eventing and bubbling infrastructure already exists. 

Would just need to add `this.fire( 'error', errorInfo )` at the right places. In the `log` module [here](https://github.com/ractivejs/ractive/blob/dev/src/utils/log.js#L83-L86) would be one place, probably some others.

**Comment by Crisfole** [comment]
Would this include things like 'printWarning' and 'printWarningOnce'?

**Comment by martypdx** [comment]
> Would this include things like 'printWarning' and 'printWarningOnce'?

I wouldn't think so, what's your thinking?

**Comment by Crisfole** [comment]
A lot of loggers, like Rollbar in particular, but also nLog, log4j, etc. etc. include the ability to log at different levels, including warnings.

It might make sense to either a) fire error and add a level parameter, or b) fire 'warn' for those.

**Comment by MartinKolarik** [comment]
I think it makes sense to have to different events for errors and warnings.

**Comment by pateketrueke** [comment]
Is there any progress on this?
